Crash loading media library on gingerbread device

```
01-24 14:47:50.073    2655-2669/? E/AndroidRuntime﹕ FATAL EXCEPTION: AsyncTask #2
    java.lang.RuntimeException: An error occured while executing doInBackground()
            at android.os.AsyncTask$3.done(AsyncTask.java:200)
            at java.util.concurrent.FutureTask$Sync.innerSetException(FutureTask.java:274)
            at java.util.concurrent.FutureTask.setException(FutureTask.java:125)
            at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:308)
            at java.util.concurrent.FutureTask.run(FutureTask.java:138)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1088)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:581)
            at java.lang.Thread.run(Thread.java:1019)
     Caused by: java.lang.IllegalArgumentException: the bind value at index 22 is null
            at android.database.sqlite.SQLiteProgram.bindString(SQLiteProgram.java:237)
            at android.database.sqlite.SQLiteDatabase.updateWithOnConflict(SQLiteDatabase.java:1721)
            at android.database.sqlite.SQLiteDatabase.update(SQLiteDatabase.java:1656)
            at org.wordpress.android.WordPressDB.saveMediaFile(WordPressDB.java:1553)
            at org.xmlrpc.android.ApiHelper$SyncMediaLibraryTask.doInBackground(ApiHelper.java:484)
            at org.xmlrpc.android.ApiHelper$SyncMediaLibraryTask.doInBackground(ApiHelper.java:412)
            at android.os.AsyncTask$2.call(AsyncTask.java:185)
            at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:306)
            at java.util.concurrent.FutureTask.run(FutureTask.java:138)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1088)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:581)
            at java.lang.Thread.run(Thread.java:1019)
```

--- a/src/org/wordpress/android/WordPressDB.java
+++ b/src/org/wordpress/android/WordPressDB.java
@@ -1540,7 +1540,7 @@ public class WordPressDB {
             boolean isMarkedForDelete = false;
             if (mf.getMediaId() != null) {
                 Cursor cursor = db.rawQuery("SELECT uploadState FROM " + MEDIA_TABLE + " WHERE mediaId=?",
-                        new String[]{mf.getMediaId()});
+                        new String[]{StringUtils.notNullStr(mf.getMediaId())});
                 if (cursor != null && cursor.moveToFirst()) {
                     isMarkedForDelete = "delete".equals(cursor.getString(0));
                     cursor.close();
@@ -1548,7 +1548,7 @@ public class WordPressDB {
 
                 if (!isMarkedForDelete)
                     result = db.update(MEDIA_TABLE, values, "blogId=? AND mediaId=?",
-                            new String[]{mf.getBlogId(), mf.getMediaId()});
+                            new String[]{StringUtils.notNullStr(mf.getBlogId()), StringUtils.notNullStr(mf.getMediaId())});
             }
 
             if (result == 0 && !isMarkedForDelete) {

